zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java

```diff
@@ -144,19 +173,24 @@ public static Either<ProblemDetail, UpdateJobRequest> toJobUpdateRequest(
                 getLongOrZero(updateRequest, r -> updateRequest.getChangeset().getTimeout())));
   }
 
-  public static CompletableFuture<ResponseEntity<Object>> executeServiceMethod(
-      final Supplier<CompletableFuture<?>> method, final Supplier<ResponseEntity<Object>> result) {
+  public static <BrokerResponseT, RestResponseT>
+      CompletableFuture<ResponseEntity<RestResponseT>> executeServiceMethod(
```

Comment:
The way I see it a `ProblemDetail` is a REST response and because of that this is more specific to me than putting `Object` in there. If I put that in there I may aswell return an Either or whatever. The fact that it's not the happy-path response doesn't mean it's not a response 😄

Reply:
Oh hold on I misread. You're saying the target REST response. As in the function we pass returns a MessageCorrelationResponse, and that's not fair because it could be a ProblemDetail

Reply:
I'm not arguing aginst that 😄 But here, the `RestResponseT` refers to the type returned by the `result` function. That function returns the "success" REST response type. Which is not the `ProblemDetail`. That might just be confusing, looking at this method and wondering how the successful REST response type will always be returned when there might be a failure case that returns something else. It's a purely theoretical discussion since Java doesn't care about the generic at runtime anyway 🤷😅

Reply:
> Oh hold on I misread. You're saying the target REST response. As in the function we pass returns a MessageCorrelationResponse whcih we use as type, and that's not fair because it could be a ProblemDetail

Yeah exactly, saw that too late 😅

Reply:
Okay, I've fixed the generics. I probably can't merge the PR yet due to the issues with the merge queue.